quickwit/quickwit-indexing/src/source/nats_source.rs

```diff
@@ -62,8 +127,40 @@ impl Source for NatsSource {
         doc_processor_mailbox: &Mailbox<DocProcessor>,
         ctx: &SourceContext
     ) -> Result<Duration, ActorExitStatus> {
+        let now = Instant::now();
+        let mut batch = BatchBuilder::default();
+        let deadline = tokio::time::sleep(*quickwit_actors::HEARTBEAT / 2);
+        tokio::pin!(deadline);
+
+        let mut subscriber = match self.client.subscribe(self.topic.into()).await {
```

Comment:
we already have the subscriber in `self.subscriber` It should not be created everytimes (there is some overhead in creating subscriber)

Reply:
yeah,this was midnight typo ~